jupyter_ai_router/router.py

```diff
@@ -157,10 +167,73 @@ def _route_message(self, room_id: str, message: Message) -> None:
 
         # Check if it's a slash command
         if first_word and first_word.startswith("/"):
-            self._notify_slash_cmd_observers(room_id, message)
+            # Extract command and create trimmed message
+            parts = message.body.split(None, 1)  # Split into max 2 parts
+            command = parts[0] if parts else ""
+            trimmed_body = parts[1] if len(parts) > 1 else ""
+            
+            # Create a copy of the message with trimmed body (command removed)
+            trimmed_message = replace(message, body=trimmed_body)
+            
+            # Remove forward slash from command for cleaner API
+            clean_command = command[1:] if command.startswith("/") else command
+            
+            # Route to slash command observers
+            self._notify_slash_cmd_observers(room_id, trimmed_message, command, clean_command)
         else:
             self._notify_msg_observers(room_id, message)
 
+    def _command_matches(self, command: str, pattern: str) -> bool:
+        """
+        Check if a command matches a pattern.
+        
+        Args:
+            command: The actual command with slash (e.g., "/help")
+            pattern: The pattern to match against without slash (e.g., "help", "ai-*", regex)
+            
+        Returns:
+            True if the command matches the pattern
+        """
+        # Convert pattern to include slash for matching
+        # Pattern "help" should match command "/help"
+        if not pattern.startswith("/"):
+            full_pattern = "/" + pattern
+        else:
+            # Handle case where pattern accidentally includes slash
+            full_pattern = pattern
+            
+        # Exact match
+        if command == full_pattern:
+            return True
+            
+        # Wildcard pattern (convert to regex)
+        if "*" in full_pattern:
+            # Escape special regex chars except *, then convert * to .*
+            escaped_pattern = re.escape(full_pattern).replace(r"\*", ".*")
+            regex_pattern = f"^{escaped_pattern}$"
+            try:
+                return bool(re.match(regex_pattern, command))
+            except re.error:
+                return False
+        
+        # Try as regex pattern (add slash if not present)
+        try:
+            return bool(re.match(full_pattern, command))
+        except re.error:
+            return False
+
+    def _notify_slash_cmd_observers(self, room_id: str, message: Message, original_command: str, clean_command: str) -> None:
```

Comment:
Since `_command_matches()` already prepends `/` to `original_command` if it does not start with `/`, can this function be simplified to just take `clean_command`?